### PR TITLE
Support @IgnoredOnParcel

### PR DESCRIPTION
**Pull request by vganin**
Adding support for `kotlinx.parcelize.IgnoredOnParcel` annotation.

### CODE REVIEW

File: parcelable/src/commonMain/kotlin/com/arkivanov/essenty/parcelable/IgnoredOnParcel.kt [hunk 1]
```diff
@@ -0,0 +1,9 @@
+package com.arkivanov.essenty.parcelable
+
+/**
+ * A multiplatform alias for `kotlinx.parcelize.IgnoredOnParcel` from the `kotlin-parcelize` Gradle plugin.
+ * Make sure you have the `kotlin-parcelize` Gradle plugin applied.
+ */
+@Target(AnnotationTarget.PROPERTY)
+@Retention(AnnotationRetention.SOURCE)
+expect annotation class IgnoredOnParcel
```
Comment (arkivanov) [inline parcelable/src/commonMain/kotlin/com/arkivanov/essenty/parcelable/IgnoredOnParcel.kt:9]: Let's annotate it also with the following, so there is no need to add the empty annotation in the `nonAndroidMain` source set. See `Parceler` or `Parcelize` for an example.

```kotlin
@OptIn(ExperimentalMultiplatform::class)
@OptionalExpectation
```

File: parcelable/src/androidTest/kotlin/com/arkivanov/essenty/parcelable/AndroidParcelableContainerRobolectricTest.kt [hunk 2]
```diff
@@ -23,6 +24,19 @@ class AndroidParcelableContainerRobolectricTest {
         assertEquals(value, consumedValue)
     }
 
+    @Test
+    fun WHEN_parcelling_ignored_THEN_returns_default_value() {
```
Comment (arkivanov) [inline parcelable/src/androidTest/kotlin/com/arkivanov/essenty/parcelable/AndroidParcelableContainerRobolectricTest.kt:28]: This class is for `ParcelableContainer` tests. It looks like `IgnoredOnParcel` could be tested separately in its own `IgnoredOnParcelTest` class. Just save the `Ignored` object into `Parcel`, and then restore it.

**Review by arkivanov — CHANGES_REQUESTED** [review]

**Comment by vganin** [comment]
I think the new annotation is also missing a constructor. Let me figure that out.

**Review by arkivanov — APPROVED** [review]

**Comment by arkivanov** [comment]
Thanks for your contribution!

**Pull request merged**